src/seedu/addressbook/data/person/Block.java

```diff
@@ -0,0 +1,35 @@
+package seedu.addressbook.data.person;
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Block {
+    public static final String EXAMPLE = "123c";
+    public static final String MESSAGE_BLOCK_CONSTRAINTS = 
+            "Person block should consist of a series of numbers, sometimes follwed by a letter";
+    public static final String BLOCK_VALIDATION_REGEX = "[\\d]*[\\w]";
+    
+    private String block;
+    
+    /**
+     * Validates given block.
+     *
+     * @throws IllegalValueException if given block string is invalid.
+     */
+    public Block(String block) throws IllegalValueException {
+       
+        if (!isBlockValid(block)) {
+            throw new IllegalValueException(MESSAGE_BLOCK_CONSTRAINTS);
+        }
+        this.block = block;
```

Comment:
It's good that you tried exceptions, but think about what prof taught about "when to use/not to use exceptions?".
And it's unusual to throw exceptions at a constructor. :) Validation of user input should be handle at a UI level by providing error messages to the user.